### DISCUSSION TITLE
request Nicegui Builder drag and drop like FLUTTERFLOW

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#688] [Ideas / Feature Requests]
If possible in the future there is a Builder tool like FLUTTERFLOW . by designing and automatically being able to copy the code results as well And there is integration to databases such as mysql and sqlite. So it can be done without coding.

**Comment by falkoschindler** [comment]
To be honest, I don't see such a graphic builder tools at the horizon for NiceGUI. Working on something like a programmatic layout builder (#364) shows how complicated it is to build user interfaces, let alone via drag-and-drop.

But nevertheless, I could imagine a project building on top of NiceGUI and providing a drag-and-drop UI generator, at least for a subset of NiceGUI's features. [CatDesign](https://github.com/ofenbach/CatDesign) from @ofenbach is an example for such a project that is based on NiceGUI but provides its own API and design language.

**Reply by v479038280** [reply]
we could use this library to make ui similar to drag and drop [nicegui-tookit](https://github.com/CrystalWindSnake/nicegui-toolkit)

Here is a brief example :
```py
from nicegui import ui, run
from niceguiToolkit.layout import inject_layout_tool

inject_layout_tool()

ui.image.default_props('fit="contain"')

def build_main():
    with ui.card().style("width:auto"), ui.row():
        ui.avatar("home")

        with ui.row():
            with ui.row():
                ui.label("data universe")

                with ui.row():
                    ui.icon("mail")
                    ui.label("send mail")

            with ui.row().style("flex-direction:column;align-self:flex-end"):
                ui.button("charge")
                ui.button("+ like 670")

with ui.row().style("width:100%;gap:10.7rem"):
    build_main()

    # with ui.row().style("align-self:center"):
    #     ui.image(rf"temp\docs\org.png").classes("w-[20rem] h-[10rem]")
    #     ui.image(rf"temp\docs\org_border.png")

ui.run()
```
It could be great if nicegui could integrate such function
@falkoschindler 
@rodja @
and I recomond another libray from the auther , [ex4nicegui](https://github.com/CrystalWindSnake/ex4nicegui) ,Its method of data binding is better than NiceGUI's, as it does not constantly refresh data in the background but only executes when data is updated.
